File: apps/backend/app/schemas/strategy.py
"""Response DTOs for `/strategy/tyres` (docs/08_API_SPECIFICATION.md — "Strategy").

Only tyre degradation is built in this phase — `/strategy/undercut`,
`/strategy/overcut`, and `/strategy/simulation` all need pit-stop-timing
data more complete than Phase 4's known `fct_pitstop` gap allows (see
`app/models/gold/pitstop.py`), and `/strategy` itself ("season strategies")
isn't well-defined beyond what `/races/{id}/strategy` and this endpoint
already cover.
"""

from __future__ import annotations

from app.schemas.base import CamelModel


class TyreDegradationPoint(CamelModel):
    compound: str
    tyre_life: float
    average_lap_time: float
    sample_count: int


class TyreDegradation(CamelModel):
    points: list[TyreDegradationPoint]

File: packages/shared-types/src/domain/strategy.ts
/**
 * `/strategy/tyres` (docs/08_API_SPECIFICATION.md — "Strategy"). Mirrors
 * `apps/backend/app/schemas/strategy.py` exactly. Only tyre degradation
 * is built on the backend — `/strategy/undercut`, `/strategy/overcut`,
 * and `/strategy/simulation` all need pit-stop-timing data more complete
 * than the known `fct_pitstop` gap allows (see
 * `apps/backend/app/models/gold/pitstop.py`), so there are no types for
 * them here.
 */

export interface TyreDegradationPoint {
  compound: string;
  tyreLife: number;
  averageLapTime: number;
  sampleCount: number;
}

export interface TyreDegradation {
  points: TyreDegradationPoint[];
}
